shark_turbine/kernel/wave/iree_utils.py

```diff
@@ -0,0 +1,54 @@
+import torch
+from .utils import compile_and_invoke
+
+
+def get_mmt_asm(lhs_type: str, rhs_type: str, acc_type: str) -> str:
+    acc_dtype = acc_type.split("x")[-1]
+    matmul_function = (
+        f"func.func @mmt(%lhs: tensor<{lhs_type}>, %rhs: tensor<{rhs_type}>) -> tensor<{acc_type}> {{\n"
+        f"  %c0 = arith.constant 0.0 : {acc_dtype}\n"
+        f"  %init = tensor.empty() : tensor<{acc_type}>\n"
+        f"  %inital_result = linalg.fill ins(%c0 : {acc_dtype}) outs(%init : tensor<{acc_type}>) -> tensor<{acc_type}>\n"
+        f"  %result = linalg.matmul_transpose_b ins(%lhs, %rhs: tensor<{lhs_type}>, tensor<{rhs_type}>)\n"
+        f"             outs(%inital_result: tensor<{acc_type}>) -> tensor<{acc_type}>\n"
+        f"  return %result : tensor<{acc_type}>\n"
+        f"}}\n"
+    )
+    return matmul_function
+
+
+def dtype_str(dtype: torch.dtype) -> str:
+    if dtype == torch.float32:
+        return "f32"
+    elif dtype == torch.float16:
+        return "f16"
+    else:
+        raise ValueError(f"Unsupported dtype: {dtype}")
```

Comment:
NIT: Can we use https://github.com/iree-org/iree-turbine/blob/36c99a686d5790add276bae3a88aa33dfb23c939/shark_turbine/support/conversions.py#L66-L86 instead?

Reply:
sure, good find!